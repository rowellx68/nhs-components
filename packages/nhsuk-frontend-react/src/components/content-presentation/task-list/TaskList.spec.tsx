import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  MultipleSections as MultipleSectionsStories,
} from './TaskList.stories';

const Default = composeStory(DefaultStory, meta);
const MultipleSections = composeStory(MultipleSectionsStories, meta);

it('should render the TaskList component with the incomplete tasks', () => {
  const { container } = render(<Default />);
  expect(container.querySelector('.nhsuk-tag--blue')).toBeInTheDocument();
});

it('should render the TaskList component with the completed tasks', () => {
  const { container } = render(<MultipleSections />);
  expect(container.querySelector('.nhsuk-task-list__hint')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
