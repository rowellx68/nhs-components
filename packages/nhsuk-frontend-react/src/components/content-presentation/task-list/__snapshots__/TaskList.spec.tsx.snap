// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the TaskList component with the completed tasks 1`] = `
<div>
  <div>
    <h1
      class="nhsuk-heading-l"
    >
      Service name
    </h1>
    <h2
      class="nhsuk-heading-m"
    >
      Check before you start
    </h2>
    <ul
      class="nhsuk-task-list"
    >
      <li
        class="nhsuk-task-list__item nhsuk-task-list__item--with-link"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <a
            aria-describedby="task-list-item_:r5:-status"
            href="#"
          >
            Check eligibility
          </a>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--completed"
          id="task-list-item_:r5:-status"
        >
          Completed
        </div>
      </li>
      <li
        class="nhsuk-task-list__item nhsuk-task-list__item--with-link"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <a
            aria-describedby="task-list-item_:r6:-status"
            href="#"
          >
            Read declaration
          </a>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--incomplete"
        >
          <strong
            class="nhsuk-tag nhsuk-tag--blue"
            id="task-list-item_:r6:-status"
          >
            Incomplete
          </strong>
        </div>
      </li>
    </ul>
    <h2
      class="nhsuk-heading-m"
    >
      Prepare application
    </h2>
    <ul
      class="nhsuk-task-list"
    >
      <li
        class="nhsuk-task-list__item nhsuk-task-list__item--with-link"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <a
            aria-describedby="task-list-item_:r7:-status"
            href="#"
          >
            Exercise
          </a>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--completed"
          id="task-list-item_:r7:-status"
        >
          Completed
        </div>
      </li>
      <li
        class="nhsuk-task-list__item nhsuk-task-list__item--with-link"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <a
            aria-describedby="task-list-item_:r8:-status"
            href="#"
          >
            Personal health
          </a>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--completed"
          id="task-list-item_:r8:-status"
        >
          Completed
        </div>
      </li>
      <li
        class="nhsuk-task-list__item nhsuk-task-list__item--with-link"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <a
            aria-describedby="task-list-item_:r9:-status task-list-item_:r9:-hint"
            href="#"
          >
            Family health history
          </a>
          <div
            class="nhsuk-task-list__hint"
            id="task-list-item_:r9:-hint"
          >
            Details of your parents and siblings
          </div>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--incomplete"
        >
          <strong
            class="nhsuk-tag nhsuk-tag--blue"
            id="task-list-item_:r9:-status"
          >
            Incomplete
          </strong>
        </div>
      </li>
      <li
        class="nhsuk-task-list__item nhsuk-task-list__item--with-link"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <a
            aria-describedby="task-list-item_:ra:-status"
            href="#"
          >
            Smoking history
          </a>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--incomplete"
        >
          <strong
            class="nhsuk-tag nhsuk-tag--blue"
            id="task-list-item_:ra:-status"
          >
            Incomplete
          </strong>
        </div>
      </li>
      <li
        class="nhsuk-task-list__item"
      >
        <div
          class="nhsuk-task-list__item-name-and-hint"
        >
          <div>
            Blood test
          </div>
        </div>
        <div
          class="nhsuk-task-list__status nhsuk-task-list__status--cannot-start-yet"
        >
          Cannot start yet
        </div>
      </li>
    </ul>
  </div>
</div>
`;
